scripts/build.py

```diff
@@ -609,6 +607,51 @@ def build_files(source, target, file_hashes):
             print e
 
 
+def rebuild_new_files(source_path, target_path, new_files_list, file_hashes):
+    """Minifies list of  new files, removes whitespace from HTML and
+    interpolates paths in HTML to include hashes in source
+    directory and copies it to target.
+
+    Args:
+        source_path: str. Path relative to /oppia directory of directory
+            containing files and directories to be copied.
+        target_path: str. Path relative to /oppia directory of directory where
+            to copy the files and directories.
+        new_files_list: list. List of recently changed file.
+        file_hashes: dict(str, str). Dictionary of file hashes.
+    """
+    for new_file in new_files_list:
+        source_file_path = (source_path + new_file).strip()
+        target_file_path = (target_path + new_file).strip()
+        ensure_directory_exists(target_file_path)
```

Comment:
Is it a directory or a file?

Reply:
The function `ensure_directory_exists` creates the directory tree that contains the file if it doesn't exists. (If  `/backend_prod_files/templates/head/pages/base.html` is `target_file_path`, `ensure_directory_exists` makes sure that `/templates/head/pages` is created).